cmd/verrazzano-operator/main_test.go

```diff
@@ -0,0 +1,174 @@
+	"/":               allHTTPMethods,
+	"/instance":       allHTTPMethods,
+	"/applications/":  allHTTPMethods,
+	"/jobs/someJobId": allHTTPMethods,
+}
+
+// Test_registerPathHandlers tests that registerPathHandlers registers the expected routes
+// and none of the routes that we don't expect
+func Test_registerPathHandlers(t *testing.T) {
+
+	// GIVEN a path prefix of /somePrefix
+	rootRouter := mux.NewRouter().StrictSlash(true)
+
+	// WHEN path handlers are registered
+	apiRouter := registerPathHandlers(rootRouter)
+
+	// THEN assert that the expected methods for each path were registered
+	for path, expectedMethods := range expectedRequestPaths {
+		for _, expectedMethod := range expectedMethods {
+			assertExpected(t, apiRouter, expectedMethod, path)
+		}
+		// AND assert that no unexpected methods were registered
+		unexpectedMethods := getComplementaryMethods(expectedMethods)
+		for _, unexpectedMethod := range unexpectedMethods {
+			assertNotExpected(t, apiRouter, unexpectedMethod, path)
+		}
+	}
+
+	// AND assert that no unexpected paths were registered
+	for unexpectedPath, methods := range unexpectedRequestPaths {
+		for _, method := range methods {
+			assertNotExpected(t, apiRouter, method, unexpectedPath)
+		}
+	}
+}
+
+func Test_initFlags(t *testing.T) {
+	// GIVEN a set of expected string and boolean flags
+	var expectedStringFlags = map[string]string{
+		"master":                       "",
+		"kubeconfig":                   "",
+		"watchNamespace":               "",
+		"verrazzanoUri":                "",
+		"helidonAppOperatorDeployment": "",
+		"enableMonitoringStorage":      "",
+		"apiServerRealm":               "",
+	}
+	var expectedBoolFlags = map[string]bool{
+		"startController": true,
+	}
+
+	foundStringFlags := map[string]string{}
+	foundBoolFlags := map[string]bool{}
+
+	// GIVEN functions to "register" string and bool flags
+	stringVarFunc := func(p *string, name string, value string, usage string) {
+		foundStringFlags[name] = value
+	}
+	boolVarFunc := func(p *bool, name string, value bool, usage string) {
+		foundBoolFlags[name] = value
+	}
+
+	// WHEN its initFlags is called
+	initFlags(stringVarFunc, boolVarFunc)
+
+	// THEN assert that the expected command line flags are initialized
+	assert.Equal(t, len(expectedStringFlags), len(foundStringFlags))
+	assert.Equal(t, len(expectedBoolFlags), len(foundBoolFlags))
+	for expectedName, expectedValue := range expectedStringFlags {
+		assert.Equal(t, expectedValue, foundStringFlags[expectedName])
+	}
+	for expectedName, expectedValue := range expectedBoolFlags {
+		assert.Equal(t, expectedValue, foundBoolFlags[expectedName])
+	}
+}
+
+// assertNotExpected asserts that the rootRouter does NOT contain a match for the given
+// http method and path
+func assertNotExpected(t *testing.T, rootRouter *mux.Router, method string, path string) {
+	req, err := http.NewRequest(method, path, nil)
+	assert.Nil(t, err, "Unexpected error creating test http request for path "+path)
```

Comment:
You can use assert.NotError.  Also this should stop the tests since req is not valid.

Reply:
this assert.Nil behaves the same way, i.e. checks for non nil and stops the test from proceeding after the assertion failure. 
If I want a good error message, I have to invoke assert.NoError in _exactly_ the same way, with the same args. Not sure why that's better.

Reply:
That is true for many of the asserts, like Len.  I think you get a better canned message if you use the more specific assert, not something you need to change.

Reply:
This is assert.Error output
        	Error Trace:	deep_test.go:30
        	Error:      	Received unexpected error:
        	            	Something error
        	Test:       	TestS1
        	Messages:   	abc failed

This is assert.Nil, not much difference
        	Error Trace:	deep_test.go:30
        	Error:      	Expected nil, but got: &errors.errorString{s:"Something error"}
        	Test:       	TestS1
        	Messages:   	abc failed

Reply:
changed to use assert.NoError